api/apiclient.go

```diff
@@ -587,22 +587,20 @@ func newWebsocketDialer(cfg *websocket.Config, opts DialOpts) func(<-chan struct
 				return nil, parallel.ErrStopped
 			default:
 			}
-			logger.Infof("dialing %q", cfg.Location)
+			logger.Debugf("dialing %q", cfg.Location)
 			conn, err := opts.DialWebsocket(cfg)
 			if err == nil {
+				logger.Debugf("successfully dialed %q", cfg.Location)
 				return conn, nil
 			}
-			if isX509Error(err) {
+			if certErr := isX509Error(err); !a.HasNext() || certErr {
 				// We won't reconnect when there's an X509
 				// error because we're not going to succeed if
 				// we retry in that case.
-				logger.Errorf("certificate error dialing %q: %v", cfg.Location, err)
-				return nil, errors.Annotatef(err, "X509 certficate error on API")
-			}
-			if !a.HasNext() {
-				logger.Errorf("error dialing %q: %v", cfg.Location, err)
+				logger.Debugf("error dialing %q (certificate error %v): %v", certErr, err)
```

Comment:
what do you think about:
detail := ""
if isCertErr {
  detail = " (bad certificate)"
}
logger.Debugf("error dialing %q%s: %v", cfg.Location, detail, err)